app/grandchallenge/components/views.py

```diff
@@ -472,3 +494,114 @@ def get(self, request, *args, **kwargs):
+                    "widget": UserUploadSingleWidget().get_context(
+                        name=interface,
+                        value=None,
+                        attrs={
+                            "id": interface,
+                            "help_text": _join_with_br(
+                                help_text if help_text else None,
+                                file_upload_text,
+                            ),
+                        },
+                    )["widget"],
+                },
+            )
+            return HttpResponse(html_content)
+        elif (
+            widget_choice == FileWidgetChoices.FILE_SELECTED.name
+            and current_value
+            and (
+                ComponentInterfaceValue.objects.filter(
+                    pk=current_value
+                ).exists()
+                if current_value.isdigit()
+                else UserUpload.objects.filter(pk=current_value).exists()
+            )
+        ):
+            # this can happen on the display set update view or redisplay of
+            # form upon validation, where one of the options is the current
+            # image, this enables switching back from one of the above widgets
+            # to the chosen image. This make sure the form element with the
+            # right name is available on resubmission.
+            html_content = render_to_string(
+                HiddenInput.template_name,
+                {
+                    "widget": {
+                        "name": interface,
+                        "value": current_value,
+                        "type": "hidden",
+                    },
+                },
+            )
+            return HttpResponse(html_content)
+        elif widget_choice == FileWidgetChoices.UNDEFINED.name:
+            # this happens when switching back from one of the
+            # above widgets to the "Choose data source" option
+            return HttpResponse()
+        else:
+            raise RuntimeError("Unknown widget type")
+
+
+class FileSearchResultView(LoginRequiredMixin, ListView):
+    template_name = "components/file_search_result_select.html"
+    search_fields = ["pk", "name"]
+    model = ComponentInterfaceValue
+    paginate_by = 50
+
+    def __init__(self, *, interface=None, user=None):
+        super().__init__()
+        self.user = user
+        self.interface = interface
+
+    def get_queryset(self):
+        return get_component_interface_values_for_user(user=self.user).filter(
+            interface=self.interface
+        )
+
+    def get(self, request, *args, **kwargs):
+        qs = self.get_queryset()
+        interface = request.GET.get("interface_slug")
+        query = request.GET.get("query-" + interface)
+        if query:
+            q = reduce(
+                or_,
+                [Q(**{f"{f}__icontains": query}) for f in self.search_fields],
+                Q(),
+            )
+            qs = qs.filter(q).order_by("name")

```

Comment:
I think this also needs changing: 
```suggestion
            qs = qs.filter(q).order_by("file__name")

```

Reply:
Same error as mentioned above. However, ordering by "file" leads to actually ordering by pk, because the pk is in the file path... I have been trying to find a solution for this, but haven't found anything yet.

Reply:
If you really wanted to, you could probably annotate the queryset with the filename and then filter on that, but I honestly think ordering by pk is fine for now.

Reply:
I've been wrecking my brain for a little while to try an annotate with the filename but I could not figure it out. I'll simply leave it at ordering by pk.